src/block-management/block-manager.js

```diff
@@ -164,14 +164,15 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	}
 
 	onChange( clientId: string, attributes: mixed ) {
-		// Update datasource UI
+		// Update Redux store
+		this.props.onChange( clientId, attributes );
```

Comment:
I pulled the onChange/redux change to before  to make act and mark the refresh variable before we go and change the block ourselfs bellow.